libsolidity/analysis/TypeChecker.cpp

```diff
@@ -780,8 +780,9 @@ bool TypeChecker::visit(Assignment const& _assignment)
 bool TypeChecker::visit(TupleExpression const& _tuple)
 {
 	vector<ASTPointer<Expression>> const& components = _tuple.components();
-	solAssert(!_tuple.isInlineArray(), "Tuple type not properly declared");
 	TypePointers types;
+	TypePointer t;
```

Comment:
It looks like your "t" variable is used only locally, it's better to declare at its closed place where it gets used.